src/automation.js

```diff
@@ -0,0 +1,97 @@
+import { rdfs, rdf } from '@tpluscode/rdf-ns-builders'
+import $rdf from '@rdfjs/dataset'
+import clownface from 'clownface'
+import TermMap from '@rdfjs/term-map'
+import * as dimensionQueries from './queries/dimensions.js'
+import * as ns from './ns.js'
+
+export async function generateDimensions(view, queries = dimensionQueries) {
+  const sources = view.out(ns.ssz.source)
+
+  clearGeneratedDimensions(view)
+
+  await createMeasureDimensions(view, sources, queries)
+  await createKeyDimensions(view, sources, queries)
+
+  return clownface({
+    dataset: $rdf.dataset([...view.dataset]),
+    term: view.term,
+  })
```

Comment:
Returns a "repackages" pointer to force re-render of the form (new JS object reference)

Reply:
This is a trick I didn't know :)

to be more explicit, I would add a method and a comment there, 

// To force re-render of the form
newReference(cf) , that simply returns cf

just because the human memory is fragile... :)

Reply:
Good idea, thanks!